ansible/playbooks/roles/repository/files/download-requirements/src/mode/base_mode.py

```diff
@@ -163,8 +187,11 @@ def __download_crane(self):
         cranes = self._requirements['cranes']
         first_crane = next(iter(cranes))  # right now we use only single crane source
 
-        downloader: Downloader = Downloader(cranes, 'sha256', self._download_crane_binary)
-        downloader.download(first_crane, crane_package_path)
+        downloader: Downloader = AltAddrDownloader(cranes, 'sha256', self._download_crane_binary)
+        if self.__check_connection(cranes[first_crane]['primary']['url']):
```

Comment:
This logic differs from that which is used for downloading files. For example, there is no warning printed here (`logging.warning('Could not connect to: `{file_to_download}`, trying secondary address...')`). How about having common method for all use cases to ensure consistency?